charts/enterprise-metrics/templates/graphite-querier-dep.yaml

```diff
@@ -0,0 +1,143 @@
+      release: {{ .Release.Name }}
+  strategy:
+    {{- toYaml .Values.graphite_querier.strategy | nindent 4 }}
+  template:
+    metadata:
+      labels:
+        app: {{ template "enterprise-metrics.name" . }}-graphite-querier
+        release: {{ .Release.Name }}
+        {{- with .Values.graphite_querier.podLabels }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+      annotations:
+{{- if .Values.useExternalConfig }}
+        checksum/config: {{ .Values.externalConfigVersion }}
+{{- else }}
+        checksum/config: {{ include (print $.Template.BasePath "/secret.yaml") . | sha256sum }}
+{{- end}}
+        {{- with .Values.graphite_querier.podAnnotations }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+    spec:
+      serviceAccountName: {{ template "enterprise-metrics.serviceAccountName" . }}
+    {{- if .Values.graphite_querier.priorityClassName }}
+      priorityClassName: {{ .Values.graphite_querier.priorityClassName }}
+    {{- end }}
+      securityContext:
+        {{- toYaml .Values.graphite_querier.securityContext | nindent 8 }}
+      initContainers:
+        {{- toYaml .Values.graphite_querier.initContainers | nindent 8 }}
+      {{- if .Values.image.pullSecrets }}
+      imagePullSecrets:
+      {{- range .Values.image.pullSecrets }}
+        - name: {{ . }}
+      {{- end}}
+      {{- end }}
+      containers:
+        - name: {{ .Chart.Name }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          args:
+            - -target=graphite-querier
+            - -config.file=/etc/enterprise-metrics/enterprise-metrics.yaml
+            - -graphite.enabled=true
+            - -graphite.querier.remote-read-enabled=true
+            - -graphite.querier.query-address=http://{{ template "enterprise-metrics.fullname" . }}-query-frontend.{{ .Release.Namespace }}.svc:{{ .Values.config.server.http_listen_port }}/api/prom
+            - -graphite.querier.graphite-fallback=http://{{ template "enterprise-metrics.fullname" . }}-graphite-web.{{ .Release.Namespace}}.svc:{{ .Values.config.server.http_listen_port }}
+            - -graphite.querier.schemas.default-storage-schemas-file=/etc/graphite-proxy/storage-schemas.conf
+            - -graphite.querier.schemas.default-storage-aggregations-file=/etc/graphite-proxy/storage-aggregations.conf
+            {{- if .Values.minio.enabled }}
+            - -admin.client.backend-type=s3
+            - -admin.client.s3.endpoint={{ .Release.Name }}-minio.{{ .Release.Namespace }}.svc:9000
+            - -admin.client.s3.bucket-name=enterprise-metrics-admin
+            - -admin.client.s3.access-key-id=enterprise-metrics
+            - -admin.client.s3.secret-access-key=supersecret
+            - -admin.client.s3.insecure=true
+            {{- end }}
+          {{- range $key, $value := .Values.graphite_querier.extraArgs }}
+            - "-{{ $key }}={{ $value }}"
```

Comment:
This checksum for config triggers a rolling upgrade whenever the configuration file changes. Do we want this behavior for changes to the graphite configuration files also?